lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -73,36 +77,158 @@ function getState() {
 	};
 }
 
-interface ImageSelectorProps {
+const URLSelector = ({ done }: { done: (imageURL: string) => void }) => {
+	const [imageURL, setImageURL] = React.useState('');
+	const [recentImages, setRecentImages]: [
+		string[],
+		(value: React.SetStateAction<string[]>) => void,
+	] = React.useState([]);
+	React.useEffect(() => {
+		const fetchRecentUrlImages = async () => {
+			try {
+				const recentUrlImages: any = await jsonStorage.get('recentUrlImages');
```

Comment:
Please avoid `any`.

Reply:
Also provide a default empty list or the code below will break.

Reply:
I couldn't find a way around the jsonStorage.get method (promisified) returning `unknown` as a type
As for the default list - it won't break, the default for that method is an empty object

Reply:
It most definitely breaks.
```
/home/alexis/dev/resin.io/etcher/node_modules/resin-corvus/src/resin-corvus.js:233 {"stack":"TypeError: recentImages is not iterable\n    at done (file:///home/alexis/dev/resin.io/etcher/generated/gui.js:4610:61)\n    at HTMLUnknownElement.callCallback (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:336:14)\n    at Object.invokeGuardedCallbackDev (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:385:16)\n    at invokeGuardedCallback (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:440:31)\n    at invokeGuardedCallbackAndCatchFirstError (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:454:25)\n    at executeDispatch (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:584:3)\n    at executeDispatchesInOrder (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:606:7)\n    at executeDispatchesAndRelease (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:713:5)\n    at executeDispatchesAndReleaseTopLevel (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:722:10)\n    at forEachAccumulated (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:694:8)\n    at runEventsInBatch (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:739:3)\n    at runExtractedPluginEventsInBatch (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:880:3)\n    at handleTopLevel (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:5807:5)\n    at batchedEventUpdates$1 (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:24368:12)\n    at batchedEventUpdates (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:1419:12)\n    at dispatchEventForPluginEventSystem (/home/alexis/dev/resin.io/etcher/node_modules/react-dom/cjs/react-dom.development.js:5898:5)","message":"recentImages is not iterable"}
```

Reply:
![Capture d’écran de 2020-04-02 16-49-01](https://user-images.githubusercontent.com/180331/78263307-102d2580-7502-11ea-8c9b-80d225d0981a.png)

Reply:
For the return type, you'll need to add a typings file for electron-json-storage in the `typings` folder.

Reply:
I was able to reproduce, it's because of the default object returned. Adding a check there

Reply:
For the return type: there are already the official typings, that's what `promisify` uses. I'll just ditch that with a custom promise, I'd rather have correct typings

Reply:
right, I haven't seen @types/electron-json-storag